src/MobSpawner.cpp

```diff
@@ -467,6 +463,18 @@ std::set<eMonsterType> cMobSpawner::GetAllowedMobTypes(EMCSBiome a_Biome)
 
 cMonster * cMobSpawner::TryToSpawnHere(cChunk * a_Chunk, Vector3i a_RelPos, EMCSBiome a_Biome, int & a_MaxPackSize)
 {
+	// If too close to any player, don't spawn anything
+	auto AbsPos = a_Chunk->RelativeToAbsolute(a_RelPos);
+	static const double RangeLimit = 24;
+	if (a_Chunk->GetWorld()->DoWithNearestPlayer(AbsPos, RangeLimit, [](cPlayer & a_Player) -> bool
+	{
+		return true;
+	})
+			)
+	{
+		return nullptr;
+	}
```

Comment:
Is this right? I thought players had to be close to mob spawners for them to work, and not the opposite?

Reply:
This is the code generating monsters independent from the mob spawner blocks.

The section you marked is just moved there from the method that checks if a mob could spawn "there" at that place. Which makes no sense as this code is shared by the natural generation and the mob spawner block. The natural generation checks now if there is a player before checking a mob can spawn at that position

Reply:
Derp, this is fine!